public/app/features/datasources/components/picker/DataSourceModal.tsx

```diff
@@ -60,6 +60,7 @@ export function DataSourceModal({
             <DataSourceList
               dashboard={false}
               mixed={false}
+              variables
```

Comment:
This is the independent list of the modal, that shows everything except the built-in DS.